module/VuFind/src/VuFind/View/Helper/Root/MakeTag.php

```diff
@@ -0,0 +1,144 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * Make tag view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeTag extends \Laminas\View\Helper\AbstractHelper
+{
+    /**
+     * List of all void tags (tags that access no innerHTML)
+     *
+     * Source: https://html.spec.whatwg.org/multipage/syntax.html#void-elements
+     *
+     * @var string[]
+     */
+    protected $voidElements = [
+        'area',
+        'base',
+        'br',
+        'col',
+        'embed',
+        'hr',
+        'img',
+        'input',
+        'link',
+        'meta',
+        'param',
+        'source',
+        'track',
+        'wbr'
+    ];
+
+    /**
+     * Render an HTML tag
+     *
+     * A string passed into $attrs will be treated like a class.
+     * These two are equivalent:
+     * > MakeTag('div', 'Success!', 'alert alert-success')
+     * > MakeTag('div', 'Success!', ['class => 'alert alert-success'])
+     *
+     * Additional options
+     * - escapeContent: Default true, set to false to skip escaping (like for HTML).
+     *
+     * @param string       $tagName  Element tag name
+     * @param string       $contents Element contents (must be properly-formed HTML)
+     * @param string|array $attrs    Tag attributes (associative array or class name)
+     * @param array        $options  Additional options
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke(
+        string $tagName,
+        string $contents,
+        $attrs = [],
+        $options = []
+    ) {
+        // $attrs not an object, interpret as class name
+        if (!is_array($attrs)) {
+            $attrs = !empty($attrs) ? ['class' => $attrs] : [];
+        }
+
+        // Compile attributes
+        return $this->compileTag($tagName, $contents, $attrs, $options);
+    }
+
+    /**
+     * Turn associative array into a string of attributes in an anchor
+     *
+     * Additional options
+     * - escapeContent: Default true, set to false to skip escaping (like for HTML).
+     *
+     * @param string $tagName   HTML tag name
+     * @param string $innerHtml InnerHTML
+     * @param array  $attrs     Tag attributes (associative array)
+     * @param array  $options   Additional options
+     *
+     * @return string
+     */
+    protected function compileTag(
+        string $tagName,
+        string $innerHtml,
+        array $attrs,
+        $options = []
+    ) {
+        $escAttr = $this->getView()->plugin('escapeHtmlAttr');
+
+        $html = '<' . $tagName;
+        foreach ($attrs as $key => $val) {
+            $html .= ' ' . $key;
```

Comment:
It would probably be good to also escape the key here. Or alternatively use the laminas-view [HtmlAttributes helper](https://docs.laminas.dev/laminas-view/helpers/html-attributes/) or `HtmlAttributesSet` directly for processing attributes.

Also, just as a general thought, this is a good example of a low-level and non-VuFind specific helper, which we maybe could consider proposing for inclusion in laminas-view? As VuFind is heavily based on Laminas, we have an interest in keeping the framework alive when it is facing competition from the more popular Laravel and Symfony. From my experience the Laminas folks are very open to contributions, respond quickly, and would be delighted with more involvement.

Reply:
And the tag name should also probably be escaped too.

Reply:
The Laminas contribution suggestion is a good point, @aleksip. It certainly wouldn't hurt to open a PR to propose a new helper and see what feedback we get, since we're not heavily invested in using this new helper in VuFind yet and could easily adapt if things evolve... and if Laminas doesn't want it, then we can resume working on it here.